transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollStreamChannel.java

```diff
@@ -365,6 +494,32 @@ protected void shutdownOutput0(final ChannelPromise promise) {
         }
     }
 
+    @Override
+    protected void doClose() throws Exception {
+        try {
+            // Calling super.doClose() first so splceTo(...) will fail on next call.
+            super.doClose();
+            if (pipeIn != -1) {
```

Comment:
Should we still clean the `pipeIn/Out` up when `super.doClose()` throws an exception?